Automattic/jetpack
# AI Assistant: implement JWT token storing

This PR adds a layer to the WJT token library. The idea is to avoid the requests.
It defines the expiration time to limit the lifetime of it.

Fixes #

## Proposed changes:

* AI Assistant: implement JWT token storing

### Other information:



## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Create an AI Assistant block
* Request a suggestion
* Confirm, the first time, the token is generated:

<img width="380" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/a74e00a3-9fbc-499e-a694-6eafca157286">

<img width="602" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/9178b503-ca1e-4236-abbe-212e7e612922">

* Perform other requests.
* Confirm that it uses the stored token, avoiding requesting a new one, if the request is made before to 2 minutes

<img width="399" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/548051fb-a6eb-4fce-8443-301652a6a9d2">

* Confirm after 2 minutes, the app requests a new token

<img width="296" alt="image" src="https://github.com/Automattic/jetpack/assets/77539/aad32338-493e-44cc-ad2f-e88f11910a7d">
